### PR TITLE
Add overflow support for the hero items in the BottomCommandingController

### PR DESCRIPTION
**Pull request by huwilkes**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Adds support for more than 5 hero items in the BottomCommandingController. The first 5 will display the same as before, but the rest will show at the top of the table view, and will be scrollable. We also deleted the +/- images we were using for the buttons, so I've added those back, along with updating the default state of the demo to show the four combinations of `isOn` x `isEnabled` hero items to make it easier to visually validate changes.

### Verification

<details>
<summary>Visual Verification</summary>

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| BottomCommandingOverflow_iPhone_Rest_Before | BottomCommandingOverflow_iPhone_Rest_After |
| BottomCommandingOverflow_iPhone_Expanded_Before | BottomCommandingOverflow_iPhone_Expanded_After |
| n/a | BottomCommandingOverflow_iPhone_Expanded_Full_After |
| BottomCommandingOverflow_iPhone_Expanded_Landscape_Before | BottomCommandingOverflow_iPhone_Expanded_Landscape_After |
| BottomCommandingOverflow_iPad_Rest_Before |  BottomCommandingOverflow_iPad_Rest_After|
| BottomCommandingOverflow_iPad_Expanded_Before | BottomCommandingOverflow_iPad_Popover_After |
</details>

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

### CODE REVIEW

File: ios/FluentUI/Bottom Commanding/BottomCommandingController.swift [hunk 1]
```diff
@@ -869,9 +959,9 @@ open class BottomCommandingController: UIViewController {
     }
 
     private class HeroItemBindingInfo: ItemBindingInfo {
-        let widthConstraint: NSLayoutConstraint
+        let widthConstraint: NSLayoutConstraint?
```
Comment (laminesm) [inline ios/FluentUI/Bottom Commanding/BottomCommandingController.swift:962]: Why is this optional now?

File: ios/FluentUI/Bottom Commanding/BottomCommandingController.swift [hunk 2]
```diff
@@ -906,7 +998,7 @@ open class BottomCommandingController: UIViewController {
         struct BottomSheet {
             static let headerHeight: CGFloat = 66
             static let headerTopMargin: CGFloat = 8
-            static let headerLeadingTrailingMargin: CGFloat = 8
+            static let headerLeadingTrailingMargin: CGFloat = 16
```
Comment (laminesm) [inline ios/FluentUI/Bottom Commanding/BottomCommandingController.swift:1001]: Should we use global tokens?

File: ios/FluentUI/Bottom Commanding/BottomCommandingController.swift (hunk 1 above)
Comment (huwilkes) [reply ios/FluentUI/Bottom Commanding/BottomCommandingController.swift:962]: Having it be non-optional meant we were applying the constraints to the overflow hero items, which was leading to constraint errors. Now we can pass in `nil`, and the `.compactMap` in `heroCommandWidthConstraints` will only return the non-nil constraints for the hero items showing in the tab bar.

File: ios/FluentUI/Bottom Commanding/BottomCommandingController.swift (hunk 2 above)
Comment (huwilkes) [reply ios/FluentUI/Bottom Commanding/BottomCommandingController.swift:1001]: I'll be working on tokenizing the control after this, so I figured just regular floats for now.

**Review by laminesm — APPROVED** [review]

**Pull request merged**